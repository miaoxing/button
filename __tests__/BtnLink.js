import "core-js/stable";
import "regenerator-runtime/runtime";

import React from 'react';
import {mount} from 'enzyme';
import {BtnLink} from '../';
import {MemoryRouter} from 'react-router';

describe('BtnLink', () => {
  test('renders correctly', () => {
    expect(mount(<MemoryRouter>
      <BtnLink href="foo">bar</BtnLink>
    </MemoryRouter>).find(BtnLink)).toMatchSnapshot();
  });

  test('text', () => {
    const wrapper = mount(<MemoryRouter>
      <BtnLink href="foo">bar</BtnLink>
    </MemoryRouter>);

    expect(wrapper.find('a').text()).toBe('bar');
  });
});
