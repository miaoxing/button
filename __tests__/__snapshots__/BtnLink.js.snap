// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnLink renders correctly 1`] = `
<BtnLink
  href="foo"
>
  <Button
    active={false}
    as={[Function]}
    disabled={false}
    href="foo"
    variant="primary"
  >
    <SafeAnchor
      as={[Function]}
      className="btn btn-primary"
      disabled={false}
      href="foo"
    >
      <Link
        className="btn btn-primary"
        href="foo"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <Link
          className="btn btn-primary"
          href="foo"
          onClick={[Function]}
          onKeyDown={[Function]}
          to={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": Object {
                "modal": undefined,
              },
            }
          }
        >
          <LinkAnchor
            className="btn btn-primary"
            href="/"
            navigate={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            <a
              className="btn btn-primary"
              href="/"
              onClick={[Function]}
              onKeyDown={[Function]}
            >
              bar
            </a>
          </LinkAnchor>
        </Link>
      </Link>
    </SafeAnchor>
  </Button>
</BtnLink>
`;
